### PR TITLE
Workflow cleanup: move checkout to start, prune unused steps+vars.

### PR DESCRIPTION
**Pull request by ScottTodd**
Cutting down on boilerplate and standardizing some workflow syntax before deeper refactoring work.

* Move `users: actions/checkout` to the start of each job (for consistency, since all workflows use the source code but not all workflows need dependencies or python setup, etc.)
* Per https://github.com/actions/checkout, `submodules: false` is the default
* Step names can be omitted with `uses` (docs: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_iduses) . Our [`pre-commit.yml`](https://github.com/nod-ai/shark-ai/blob/main/.github/workflows/pre-commit.yaml) workflow does this already: https://github.com/nod-ai/shark-ai/blob/c0ca2e2788ea09a890ffe347dfbffe57606670b8/.github/workflows/pre-commit.yaml#L11-L14
image
* Remove copy/pasted `Get Current Date` step from some workflows
* Remove unused `SHARK_PLATFORM_REPO_ROOT` env var from some workflows (unused since https://github.com/nod-ai/shark-ai/pull/305, copy/pasted into a few places)

### CODE REVIEW

**Review by stbaione — APPROVED** [review]
LGTM, test failures for `perplexity short` and `Llama Benchmarking` are due to #638, not anything with this PR

File: .github/workflows/ci_eval.yaml [hunk 1]
```diff
@@ -36,17 +36,15 @@ jobs:
         shell: bash
     env:
       PIP_CACHE_DIR: "${{ github.workspace }}/.pip-cache"
-      SHARK_PLATFORM_REPO_ROOT: ${{ github.workspace }}
     steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```
Comment (archana-ramalingam) [inline .github/workflows/ci_eval.yaml:40]: Just seeing this now, there is another job in this workflow that needs to be updated as well. Maybe in another PR.

File: .github/workflows/ci-sharktank.yml [hunk 2]
```diff
@@ -36,15 +36,14 @@ jobs:
     env:
       PIP_CACHE_DIR: "${{ github.workspace }}/.pip-cache"
     steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
```
Comment (archana-ramalingam) [inline .github/workflows/ci-sharktank.yml:40]: Same here, there is another job in this workflow that needs to be updated as well.

**Comment by ScottTodd** [comment]
Ah. Thanks for noticing. Sent https://github.com/nod-ai/shark-ai/pull/642

**Review by marbre — COMMENTED** [review]
Thanks for the cleanup.

**Pull request merged**